towhee/functional/mixins/parallel.py

```diff
@@ -233,11 +235,17 @@ def inner():
 
         async def worker():
             buff = []
-            for x in self:
+            iterable = self._iterable.chunks() if isinstance(self._iterable, ChunkedTable) else self
+            for x in iterable:
                 if len(buff) == num_worker:
                     queue.put(await buff.pop(0))
                 buff.append(
-                    loop.run_in_executor(executor, _map_task(x, unary_op)))
+                    loop.run_in_executor(executor, self._map_task(x, unary_op)))
+            # for x in self:
+            #     if len(buff) == num_worker:
+            #         queue.put(await buff.pop(0))
+            #     buff.append(
+            #         loop.run_in_executor(executor, self._map_task(x, unary_op)))
             while len(buff) > 0:
                 queue.put(await buff.pop(0))
             queue.put(EOS())
```

Comment:
the return value of pmap also needs to be changed.

Reply:
Shouldn't it be a new dc contains Writabletable? I've changed the return value of `_map_task` accordingly, i.e. return a Writabletable for each chunk, I suppose this is the expected behavior.